commit 39e9481f761db5f78458473a88aeb237fd93b756
Author: Richard Weinberger <richard@nod.at>
Date:   Thu Mar 30 10:50:49 2017 +0200

    ubifs: Fix O_TMPFILE corner case in ubifs_link()
    
    commit 32fe905c17f001c0eee13c59afddd0bf2eed509c upstream.
    
    It is perfectly fine to link a tmpfile back using linkat().
    Since tmpfiles are created with a link count of 0 they appear
    on the orphan list, upon re-linking the inode has to be removed
    from the orphan list again.
    
    Ralph faced a filesystem corruption in combination with overlayfs
    due to this bug.
    
    Cc: Ralph Sennhauser <ralph.sennhauser@gmail.com>
    Cc: Amir Goldstein <amir73il@gmail.com>
    Reported-by: Ralph Sennhauser <ralph.sennhauser@gmail.com>
    Tested-by: Ralph Sennhauser <ralph.sennhauser@gmail.com>
    Reported-by: Amir Goldstein <amir73il@gmail.com>
    Fixes: 474b93704f321 ("ubifs: Implement O_TMPFILE")
    Signed-off-by: Richard Weinberger <richard@nod.at>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1507_39e9481f761db5f78458473a88aeb237fd93b756.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1507_39e9481f761db5f78458473a88aeb237fd93b756.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
